src/solvers/td3.ml

```diff
@@ -719,6 +719,44 @@ module WP =
             ) marked_for_deletion
         );
 
+        (* [destabilize_leaf] is meant for restarting of globals selected by the user. *)
+        (* Must be called on a leaf! *)
+        let destabilize_leaf (x : S.v) =
+          let destab_side_dep (x : S.v) =
+            match HM.find_option side_dep x with
+            | Some w ->
+              HM.remove side_dep x;
+              VS.iter (fun y ->
+                  HM.remove stable y;
+                  destabilize y;
+                ) w;
```

Comment:
This is missing a lot of the `side_dep` logic that `destabilize_with_side` has. In particular the superstable and abort front stuff. I think that part of `destabilize_with_side` could just be copied here and removing the recursive call.